Detectors/ZDC/simulation/src/Detector.cxx

```diff
@@ -2398,6 +2430,22 @@ void Detector::BeginPrimary()
   resetHitIndices();
 
   mCurrentPrincipalParticle = *stack->GetCurrentTrack();
+
+#ifdef ZDC_FASTSIM_ONNX
+  if (o2::zdc::ZDCSimParam::Instance().useZDCFastSim && mFastSimModel != nullptr) {
+    std::vector<float> particle = {static_cast<float>(mCurrentPrincipalParticle.Energy()),
```

Comment:
So this looks like you are creating a new std::vector at each invocation which is overhead.
Consider using a different fixed-size type or struct that can be passed to the fast sim model, so that it can be constructed on the stack.

Reply:
std::vector still will be created at each invocation inside fast sim model wrapper. What I think can do, is to change interface so the setData method will get mCurrentPrincipalParticle and extract necessery data itself. What do you think about it?

PS. I do not want to create fixed size container to provide as much flexibility for future models as possible.

Reply:
Fixed in commit [3e2161d](https://github.com/AliceO2Group/AliceO2/pull/8396/commits/3e2161d5bc10386f37db67ce3d2f8ad15cb1cea1)